### ISSUE TITLE
Looking for maintainers

### ISSUE THREAD

**Issue by AlistairB** [haskell/docker-haskell#72]
I am currently the only maintainer.. :sweat_smile: 

## Useful skills

Roughly in order of usefulness.

- Knowledge of Docker
- Knowledge of Haskell tooling
- Knowledge of operating systems

However, none of these are a hard requirement. The bumping of versions is a pretty simple process. Happy to pass on my knowledge if someone is interested but hasn't done much docker or whatever.

To make it clear, there is a very low level of knowledge required to get involved.

## What do maintainers do?

Primarily, bump versions of GHC, cabal + stack when they are released. These are bumped first in this repo, then in the official images [repo](https://github.com/docker-library/official-images/blob/master/library/haskell).

Otherwise, look for ways to improve the images, or add more OS support.

## I am looking to step down

Update: I am looking to step down end of Jan 2023 (happy to stick around longer to advise if needed). I'll post this around and hopefully people are interested.

Update 2: I am around and will approve and merge PRs, but will not make any more commits in the hope that others will step up. At the latest I will leave the project end of 2023.

## Next steps

If you are interested, let me know a bit about yourself and why you are interested. Then I think you can do the next version bumps that come along, or dive into other more thorny work if you want.

**Comment by anik3t** [comment]
@AlistairB I would like to help in maintaining it.

**Comment by AlistairB** [comment]
@anik3t Awesome! Perhaps you could tell me a bit about your experience, timezone and anything else you think may be relevant? You are pretty much welcome to join and help regardless, but might be useful to understand what skills you have around docker and Haskell in terms of what I might need to teach you to help maintain the images. I'd also be curious to know why you are interested in helping and what kind of work in particular may be of interest.

For me, I live in Melbourne, Australia UTC+10/11. I've worked in a variety of languages, but done a fair amount of Scala and I am passionate about functional programming and Haskell. I created a [small github app thing](https://deadpendency.com) in Haskell and I use these images for that, which is how I got involved as I was eager to have them updated to new GHC / Stack versions.

Anyway, I think probably the best place to start would be to do the next version bump that comes along. I tried to document the full process in the readme so perhaps you could follow that and give it a go?

Otherwise, there is also work to support Alpine and Windows. The alpine work is pretty much done, just waiting on GHC to have a working 9.2.X version on Alpine. Windows has a basic proof of concept working, however there is much more work around figuring out a testing (in CI primarily) strategy (in particular Stack needs additional configuration). There are also other issues that you could help with or of course I am very open to ideas to improve the images further if you have any.

Anyway, that is a bit of a wall of text.. Glad you are interested in helping!

**Comment by jappeace** [comment]
I can help with any alpine/ubuntu based images since I use that from time to time, I've little interest in windows support right now.

**Comment by AlistairB** [comment]
Thanks @jappeace !

There isn't a lot of work required on the debian images, aside from [providing debian bullseye support](https://github.com/haskell/docker-haskell/issues/73) once Haskell tooling supports it. Although, any idea for improvement on the existing images is always welcome.

[Alpine support](https://github.com/haskell/docker-haskell/issues/22) is more or less [done](https://github.com/haskell/docker-haskell/pull/60), but blocked on ghci segfaulting ([1](https://gitlab.haskell.org/ghc/ghc/-/issues/20266), [2](https://gitlab.haskell.org/ghc/ghc/-/issues/22237)) for 9.2+. I don't think much is to be done other than wait on GHC developers to fix this. Although again, if you think the Alpine images can be improved that is useful.

**Comment by jappeace** [comment]
yes I'll do boring version bumping, I currently don't have time to fix ghci segfaults, maybe later, sounds sorta challenging.

**Comment by AlistairB** [comment]
oh, no you don't need to touch any of the upstream dependencies like GHC. They are aware of the issue it is up to them if they want to prioritize fixing. Debian bullseye is in a similar boat, we simply wait until cabal, stack + GHC support it officially then we can release support in docker.

**Comment by tarleb** [comment]
Hi, I'm interested in helping out as well. I'm maintaining the pandoc Docker images (https://github.com/pandoc/dockerfiles) and consider to restructure our build setup, which would include a switch to the official Haskell images. Helping to maintain images that we depend on would only be fair, I think. We currently support Alpine and Ubuntu, so those are the systems that I'm most interested in (and most familiar with, but not at a particularly high level of knowledge).

**Comment by jappeace** [comment]
How do I go about doing boring version bumps, is there some kind off notification system?

**Comment by AlistairB** [comment]
@jappeace in terms of cabal and stack, you can watch the releases on github.

![image](https://user-images.githubusercontent.com/437253/206581412-9e3d3ebc-7312-47b6-8614-9ccbc057c5bf.png)

For GHC I usually just know because I follow the ecosystem fairly closely. One way might be to use the RSS feed for the [GHC blog](https://www.haskell.org/ghc/blog.html).

@tarleb Awesome! Sounds good. I think next steps are to do one of the version bumps. Check out the github readme for the process. I guess one of you or @jappeace could do it or alternate or whatever. First in best dressed? ;)

**Comment by AlistairB** [comment]
So [stack 2.9.3 has been released](https://discourse.haskell.org/t/ann-stack-2-9-3/5459).

This will need to be bumped in the images. This release has a bit of an extra piece of complexity in that it is the first official stack release to support debian aarch64. Currently in the images [stack is not installed on aarch64](https://github.com/haskell/docker-haskell/issues/59). I have updated this in https://github.com/haskell/docker-haskell/pull/93 which uses the RC to test. This will need to be updated to the final release. I'd suggest just creating a new PR.

Hopefully following the readme and looking at the images you can figure this out. The main thing to do is to update the release SHA + version. If anything is not clear from the readme let me know and I'll update it to add more info.

**Comment by AlistairB** [comment]
@tarleb @jappeace are either of you interested in doing this stack version bump?

**Comment by tarleb** [comment]
Apologies, the latest wave of viruses hit pretty hard, plus trying to get pandoc 3 out the door. It seems that I won't be able to dive in before the start of next year.

**Comment by AlistairB** [comment]
No problem at all!

**Comment by jappeace** [comment]
trying, where do you find the stack_release_key?

**Comment by jappeace** [comment]
https://github.com/haskell/docker-haskell/pull/94

**Comment by AlistairB** [comment]
I added some comments to the PR @jappeace thanks. Also https://discourse.haskell.org/t/ghc-9-4-4-is-now-available/5481 is another bump that needs to happen. I'm trying to avoid doing these bumps so others can learn the ropes. I can jump in and do one if people want.

**Comment by AlistairB** [comment]
I have gone ahead and bumped stack + GHC in https://github.com/haskell/docker-haskell/pull/93 + https://github.com/docker-library/official-images/pull/13799. I'd prefer the images to not get too far behind. I'll work on improving the docs and hopefully someone else can do the next bump.

**Comment by AlistairB** [comment]
I have updated the readme with much more detailed instructions on doing the bumps. Please let me know if any of this is unclear or can be improved. Thanks

**Comment by AlistairB** [comment]
https://www.haskell.org/ghc/download_ghc_9_2_6.html

9.2.6 released if anyone is interested in doing the bump?

**Comment by jappeace** [comment]
Hi, sorry, I don't think I can manage maintenance of this.

**Comment by AlistairB** [comment]
No worries!

**Comment by AlistairB** [comment]
@tarleb how about you?

**Comment by tarleb** [comment]
:+1: I'll give it a try later today.

**Comment by AlistairB** [comment]
@tarleb are you interested / have time to do 9.2.7 too?

**Comment by tarleb** [comment]
Yes to both. I might not get to it today, but tomorrow should work.

**Comment by AlistairB** [comment]
Thanks again! If you still keen I'm happy to add you as an official maintainer? I'll probably hang around for a few months yet chipping in. After that you can still ping me if you want another opinion.

(It would be nice to get more maintainers of course too..)

**Comment by tarleb** [comment]
Slightly scared, but sounds good. Having more maintainers would be great of course; it's sometimes hard to do updates in a timely manner if too many other things came up at the same time.

**Comment by AlistairB** [comment]
Awesome! Yeah, there is no pressure to do things in a timely manner. Any help doing updates is helping out the Haskell ecosystem. If people who use the images really want an update they can get involved / submit a PR if they need (which is how I got involved).

Even if you decide to step away, ultimately the Haskell Foundation or someone will step up to maintain them. So I'd suggest giving only what time you can comfortably do.

I've added you as an admin to this repo. The other thing is to add you as a maintainer [here](https://github.com/docker-library/official-images/blob/master/library/haskell#L1). I can do if you give me what email you want to use, or you can submit a PR and I can approve.

**Comment by rickprice** [comment]
Do you still need help maintaining the docker images?

**Comment by AlistairB** [comment]
@rickprice yup definitely. If you want to get involved you could give the GHC 9.2.8 bump a go?

**Comment by rickprice** [comment]
Okay, I will try to do something in the next day or so.
I'm learning Haskell, and I've been programming for a good 40 years, we use Docker for work stuff so I'm reasonably familiar with it.

**Comment by AlistairB** [comment]
Great. Hopefully the [readme](https://github.com/haskell/docker-haskell#updating-the-images) is self-explanatory.

**Comment by AlistairB** [comment]
Ok, now seems like a reasonable time to step away @tarleb @develop7 . I've cleaned up a few last PRs / issues and removed my access. I'm going to unfollow the repo, but feel free to ping me if you have any questions. I guess I'll close this particular issue. Of course you may want to open another similar one depending on how you are going. Best of luck!

**Comment by benz0li** [comment]
@AlistairB Thank you for your commitment over the last 3.5 years.
